### PR TITLE
Update webhook push event handler.

### LINKED ISSUE #803: Webhooks: handlePush can be simplified and tests should be added
The current `web/hooks/handlePush()` should be simplified. 

- [ ] Ignore pushes to non-default branches should be checked before the switch-case: `if !isDefaultBranch(payload)...`
- [ ] Related to the first point; we need to tackle that group repositories with pull requests can push to non-default branches. So group repos needs special handling.
- [ ] `wh.updateLastActivityDate(repo.UserID, course.ID)` and the if necessary `jobOwner, err := wh.db.GetUserByCourse(course, payload.GetSender().GetLogin())` could be done before the switch case. 
- [ ] The `UserRepo` and `GroupRepo` cases could be simplified using a closure function.

### PR DESCRIPTION
**Pull request by 0xf8f8ff**
Resolves #803.

Adding relevant tests in a separate PR.

### CODE REVIEW

File: web/hooks/push.go [hunk 1]
```diff
@@ -68,48 +69,23 @@ func (wh GitHubWebHook) handlePush(payload *github.PushEvent) {
...
 		assignments := wh.extractAssignments(payload, course)
 		for _, assignment := range assignments {
-			if !assignment.IsGroupLab {
-				// only run non-group assignments
-				wh.runAssignmentTests(sc, assignment, repo, course, payload)
-			} else {
-				wh.logger.Debugf("Ignoring push to user repo %s for group assignment: %s", payload.GetRepo().GetName(), assignment.GetName())
+			results := wh.runAssignmentTests(sc, assignment, repo, course, payload)
+			// Non-default branch indicates push to a group repo.
+			if !isDefaultBranch(payload) && !assignment.GradedManually() {
```
Comment (meling) [inline web/hooks/push.go:78]: If you move this code into `runAssignmentTests` you can remove the `GradedManually()` check (it is already done in `runAssignmentTests`) and we can remove the `results` return value from `runAssignmentTests`.

Comment (meling) [reply web/hooks/push.go:78]: The loop becomes nice and tidy:
```go
for _, assignment := range assignments {
	wh.runAssignmentTests(sc, assignment, repo, course, payload)
}
```

File: web/hooks/push.go [hunk 2]
```diff
@@ -68,48 +69,23 @@ func (wh GitHubWebHook) handlePush(payload *github.PushEvent) {
...
-					wh.handlePullRequestPush(payload, results, assignment, course, repo)
-				}
-			} else {
-				wh.logger.Debugf("Ignoring push to group repo %s for user assignment: %s", payload.GetRepo().GetName(), assignment.GetName())
-			}
-		}
 	default:
 		wh.logger.Debug("Nothing to do for this push event")
 	}
+
```
Comment (meling) [inline web/hooks/push.go:88]: Unnecessary blank line

File: web/hooks/push.go (hunk 1 above)
Comment (meling) [reply web/hooks/push.go:78]: The check at the end of the `runAssignmentTests` func becomes:
```go
	// Non-default branch indicates push to a group repo.
	if !isDefaultBranch(payload) {
		// Attempt to find the pull request for the branch, if it exists,
		// and then assign reviewers to it, if the branch task score is higher than the assignment score limit
		wh.handlePullRequestPush(payload, results, assignment, course, repo)
	}
```

Comment (meling) [reply web/hooks/push.go:78]: And you can remove all the returns.

File: web/hooks/push.go [hunk 3]
```diff
@@ -27,13 +27,22 @@ func (wh GitHubWebHook) handlePush(payload *github.PushEvent) {
 	}
 	wh.logger.Debugf("Received push event for repository %v", repo)
 
+	if !(isDefaultBranch(payload) || repo.IsGroupRepo()) {
```
Comment (meling) [inline web/hooks/push.go:30]: I felt this "guard" was more difficult to read than using `!isDefaultBranch && !repo.IsGroupRepo`. I think both are correct.

**Review by meling — CHANGES_REQUESTED** [review]
This looks nice! A few minor tweaks and this should be fine, I think...

File: web/hooks/push.go (hunk 3 above)
Comment (0xf8f8ff) [reply web/hooks/push.go]: Agree, this one is harder to read. Changed to the `!isDefaultBranch && !repo.IsGroupRepo`.

**Review by meling — APPROVED** [review]

**Pull request merged**